pallets/creditcoin/src/migrations/v8.rs

```diff
@@ -0,0 +1,232 @@
+				weight = weight.saturating_add(weight_each);
+
+				match z {
+					OldTask::VerifyTransfer(pending) => {
+						let new = UnverifiedTransfer {
+							transfer: pending.transfer,
+							from_external: pending.from_external,
+							to_external: pending.to_external,
+							deadline: pending.deadline,
+						};
+
+						return Some(crate::types::Task::VerifyTransfer(new));
+					},
+					OldTask::CollectCoins(pending) => {
+						let new = UnverifiedCollectedCoins {
+							to: pending.to,
+							tx_id: pending.tx_id,
+							contract: pending.contract,
+							contract_type: crate::types::ContractType::GCRE,
+						};
+
+						return Some(crate::types::Task::CollectCoins(new));
+					},
+				}
+			},
+		);
+		return weight;
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use crate::helpers::extensions::IntoBounded;
+	use pallet_offchain_task_scheduler::tasks::TaskV2;
+
+	use super::*;
+	use crate::{
+		migrations::v6::OldUnverifiedCollectedCoins,
+		mock::{self, ExtBuilder, Test},
+		types::{test::create_unverified_transfer, ContractType},
+		Task,
+	};
+
+	#[frame_support::storage_alias]
+	type CollectedCoins<T: crate::Config> = StorageMap<
+		crate::Pallet<T>,
+		Identity,
+		CollectedCoinsId<<mock::Test as frame_system::Config>::Hash>,
+		OldCollectedCoinsStruct<
+			<mock::Test as frame_system::Config>::Hash,
+			<mock::Test as pallet_balances::Config>::Balance,
+		>,
+	>;
+
+	type OldCollectedCoinsStorage = CollectedCoins<Test>;
+
+	#[test]
+	fn test_migrate_collected_coins_struct() {
+		ExtBuilder::default().build_and_execute(|| {
+			let tx_id = BoundedVec::default();
+
+			let storage_id = crate::CollectedCoinsId::new::<crate::mock::Test>(
+				&crate::Blockchain::Ethereum,
+				&tx_id,
+			);
+
+			let address: [u8; 20] = [1; 20];
+
+			let to = AddressId::new::<Test>(&crate::Blockchain::Ethereum, &address);
+			let old = OldCollectedCoinsStruct { to, amount: 100, tx_id };
+
+			OldCollectedCoinsStorage::insert(&storage_id, &old);
+
+			super::Migration::<Test>::new().migrate();
+
+			let new = crate::CollectedCoins::<Test>::try_get(&storage_id).unwrap();
+
+			assert_eq!(old.to, new.to);
+			assert_eq!(old.amount, new.amount);
+			assert_eq!(old.tx_id, new.tx_id);
+			assert_eq!(new.contract_type, ContractType::GCRE);
+		})
+	}
+
+	#[test]
+	fn migrate_collect_coins() {
+		ExtBuilder::default().build_and_execute(|| {
+			let pending = OldUnverifiedCollectedCoins {
+				to: [0u8; 256].into_bounded(),
+				tx_id: [0u8; 256].into_bounded(),
+				contract: Default::default(),
+			};
+			let id = TaskId::CollectCoins(crate::types::CollectedCoinsId::from(
+				TaskV2::<Test>::to_id(&pending),
+			));
+
+			PendingTasks::<Test>::insert(1u64, id.clone(), OldTask::from(pending.clone()));
+
+			super::Migration::<Test>::new().migrate();
+
+			let migrated_pending = {
+				if let Task::CollectCoins(pending) = new::PendingTasks::<Test>::get(1, id).unwrap()
+				{
+					pending
+				} else {
+					unreachable!()
+				}
+			};
```

Comment:
```suggestion
			let Task::CollectCoins(migrated_pending) = new::PendingTasks::<Test>::get(1, id).unwrap()
			else {
				unreachable!()
			};
```
The semi-recently stabilized [let-else](https://doc.rust-lang.org/rust-by-example/flow_control/let_else.html) makes this less wordy